lib/process-services-cloud/src/lib/services/user-preference-cloud.service.ts

```diff
@@ -37,10 +38,19 @@ export class UserPreferenceCloudService extends BaseCloudService implements Pref
    * @param appName Name of the target app
    * @returns List of user preferences
    */
-  getPreferences(appName: string): Observable<any> {
+  getPreferences(appName: string, key?: string): Observable<any> {
     if (appName) {
       const url = `${this.getBasePath(appName)}/preference/v1/preferences`;
-      return this.get(url);
+
+      return this.get(url).pipe(map((preferences: any) => {
+        if (key) {
+          const preferencesList = preferences?.list?.entries ?? [];
+          const searchedPreferences = preferencesList.find(preference => preference.entry.key === key);
+          return searchedPreferences ? JSON.parse(searchedPreferences.entry.value) : null;
```

Comment:
This seems a little bit confusing, maybe wrap this information in a model would help getting what are we trying to populate here and will help having a single return to identify what is returned.

Reply:
It was my error (reverted it), unfortunately we are returning here `any` which does not help with inferring what we should return.

When I checked how this is implemented in `local-preference-cloud.service` (these both services implements the same interface `PreferenceCloudServiceInterface` and we should be able to use them interchangeably)

we are returning structure like:

```
{
  list: {
   entries: [ {
          entry: {
            key,
            value: 'MY VALUE'
          }
       }]
   }
};
```
Since we are using `PreferenceCloudServiceInterface` with method signature that looks like: `getPreferences(appName: string, key?: string): Observable<any>;`

I would thought that after passing the `key` e.g. `my-preferences` I am going to get preferences for that `key` and I do not have to perform search every time.

Thats why I created `searchPreferences` that is going to return exactly what we need: https://github.com/Alfresco/alfresco-ng2-components/pull/7569/files#diff-b04373b5164c2ea36b539499ce6aab959e51532f2b379cb6b65734739028a88bR27 (I will change `key: any` to list of keys if I need to get more preferences as  'key: string[]')

Reply:
I see that searchPreferences causing some confusion, therefore I removed it and used `getPreferences`.
We can think about updating the preferences service some other time :)

Reply:
If that helps the clarity we could define a model to wrap this entity

Reply:
looks like different services implements it in different way, so it should be generic, currently I will skip it since this is not in scope of this story (basically I am lack of time :confused:)

Reply:
I am getting some issues while applying this model, will leave it right now, there need to be more changes for type this properly.